crates/events/src/extend.rs

```diff
@@ -88,18 +88,18 @@ impl AttributesMap for Vec<namada_core::tendermint::abci::EventAttribute> {
         K: Into<String>,
         V: Into<String>,
     {
-        self.push(namada_core::tendermint::abci::EventAttribute {
-            key: key.into(),
-            value: value.into(),
-            index: true,
-        });
+        self.push((key, value, true).into());
     }
 
     #[inline]
     fn retrieve_attribute(&self, key: &str) -> Option<&str> {
         self.iter().find_map(|attr| {
-            if attr.key == key {
-                Some(attr.value.as_str())
+            if attr.key_str().expect("Attribute key is malformed UFT-8") == key
+            {
+                Some(
+                    attr.value_str()
+                        .expect("Attribute value is malformed UTF-8"),
+                )
```

Comment:
hm, this should be fine, as long as we always trust the input source of these events. either way, it looks like this is only fallible if someone constructs a v034 event, which contained base64 encoded strings. maybe we should fail silently, though? returning `None` on these conversion failures, and (perhaps) emitting a `tracing::debug` log msg

Reply:
thanks, it makes sense. I fixed it at 9087cac